Tests/TensorFlowTests/OptimizerTests.swift

```diff
@@ -0,0 +1,54 @@
+// Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+@testable import TensorFlow
+
+final class OptimizerTests: XCTestCase {
+    func testRAdam() {
+        struct Model: Layer {
+            var w: Tensor<Float>
+            var b: Tensor<Float>
+
+            public init(w: Tensor<Float>, b: Tensor<Float>) {
+                self.w = w
+                self.b = b
+            }
+
+            @differentiable
+            public func callAsFunction(_ input: Tensor<Float>) -> Tensor<Float> {
+                return matmul(input, w) + b
+            }
+        }
+        let w = Tensor<Float>([[1.0], [2.0]])
+        let b = Tensor<Float>([0.0])
+        var model = Model(w: w, b: b)
+        let optimizer = RAdam(for: model, learningRate: 1e-3)
+        // To obtain gradient of type Model.TangentVector 
+        var gradient = model.gradient { model -> Tensor<Float> in 
```

Comment:
Please use top-level differential operators instead of method-style ones, which are deprecated and already have been (?) removed:
```suggestion
        var gradient = gradient(at: model) { model -> Tensor<Float> in 
```
Reason for removing the method-style ones: they're confusing and duplicate the top-level ones. Let's standardize on the top-level ones.

Reply:
Sure, will work on the changes now. Just to clarify, you're talking about `valueWithGradient()` right ?

Reply:
> Sure, will work on the changes now. Just to clarify, you're talking about `valueWithGradient()` right ?

No, I'm just talking about `gradient` actually. Does the exact suggested code change (`gradient(at: model) { ... }`) make sense to you? It does the same thing as `model.gradient { ... }`, except the latter has been removed to reduce API surface area.

Reply:
Oh alright, yeah that makes sense. I'll be pushing the changes in sometime.